### PR TITLE
Add support for Sifive-PWM.

### PR DESCRIPTION
**Pull request by NandkumarJoshi**
Add example-pwm in freedom-e-sdk, with sifive-pwm driver. Updated dts, bsp files with related changes.
Related PRs https://github.com/sifive/freedom-metal/pull/232 , https://github.com/sifive/freedom-metal/pull/206 .

### CODE REVIEW

File: bsp/qemu-sifive-e31/core.dts [hunk 1]
```diff
@@ -161,7 +165,7 @@
                         reg = <0x10013000 0x1000>;
                         reg-names = "control";
                         clocks = <&hfclk>;
-                        pinmux = <&gpio0 0x30000 0x30000>;
+                        pinmux = <&gpio0 0x0 0x30000>;
```
Comment (bsousi5) [inline bsp/qemu-sifive-e31/core.dts]: what is the reason for this change? previously incorrect?

File: bsp/qemu-sifive-e31/core.dts [hunk 2]
```diff
@@ -170,7 +174,7 @@
                         reg = <0x10014000 0x1000 0x20000000 0x20000000>;
                         reg-names = "control", "mem";
                         clocks = <&hfclk>;
-                        pinmux = <&gpio0 0x0003C 0x0003C>;
+                        pinmux = <&gpio0 0x0 0x0003C>;
```
Comment (bsousi5) [inline bsp/qemu-sifive-e31/core.dts]: same here, previously incorrect?

File: bsp/sifive-hifive1-revb/core.dts [hunk 3]
```diff
@@ -167,7 +167,7 @@
                         reg = <0x10013000 0x1000>;
                         reg-names = "control";
                         clocks = <&hfclk>;
-                        pinmux = <&gpio0 0x30000 0x30000>;
+                        pinmux = <&gpio0 0x0 0x30000>;
```
Comment (bsousi5) [inline bsp/sifive-hifive1-revb/core.dts]: I assume these has been confirm with the Data Sheet?

File: bsp/qemu-sifive-e31/core.dts (hunk 1 above)
Comment (NandkumarJoshi) [reply bsp/qemu-sifive-e31/core.dts]: A change was made in gpio driver, to enable to set a bit in iof_sel register.
https://github.com/sifive/freedom-metal/blob/sifive-pwm/src/drivers/sifive_gpio0.c#L103
Previously it was, ''&=" operation, which wont allow to set a bit, if existing value is zero. 
As per GPIO Muxing for FE310, only PWM IP outputs are mapped to 'IOF1' everything else is mapped to 'IOF0'.
First value in this pinmux pair maps to 'source' parameter in __metal_driver_sifive_gpio0_enable_io() function. 
Corresponding values in dts need to be changed accordingly.

File: bsp/sifive-hifive1-revb/core.dts (hunk 3 above)
Comment (NandkumarJoshi) [reply bsp/sifive-hifive1-revb/core.dts]: As per GPIO Muxing for FE310, only PWM IP outputs are mapped to 'IOF1' everything else is mapped to 'IOF0'.
GPIO driver is changed, to allow PWM to be selected from IOF register.

https://github.com/sifive/freedom-metal/blob/sifive-pwm/src/drivers/sifive_gpio0.c#L103

File: bsp/qemu-sifive-e31/core.dts (hunk 2 above)
Comment (NandkumarJoshi) [reply bsp/qemu-sifive-e31/core.dts]: dts values changed, as per a change in gpio driver.
https://github.com/sifive/freedom-metal/pull/232

File: .gitmodules [hunk 4]
```diff
@@ -136,3 +136,6 @@
 [submodule "scripts/esdk-settings-generator"]
 	path = scripts/esdk-settings-generator
 	url = https://github.com/sifive/esdk-settings-generator.git
+[submodule "software/example-pwm"]
```
[the remark this answers is not in the document]
Comment (NandkumarJoshi) [reply .gitmodules]: fixed.

**Comment by nategraff-sifive** [comment]
Wait! This shouldn't have been merged until the submodules were bumped to point at master after they were merged!

**Comment by bsousi5** [comment]
Oh shoot. I messed that up.
